userspace/libsinsp/filter_check_list.cpp

```diff
@@ -99,27 +90,27 @@ sinsp_filter_check* filter_check_list::new_filter_check_from_fldname(const std::
 	// it's very likely that you've forgotten to add your filter to the list in
 	// the constructor
 	//
-	return NULL;
+	return nullptr;
 }
 
 sinsp_filter_check_list::sinsp_filter_check_list()
 {
 	//////////////////////////////////////////////////////////////////////////////
 	// ADD NEW FILTER CHECK CLASSES HERE
 	//////////////////////////////////////////////////////////////////////////////
-	add_filter_check(new sinsp_filter_check_gen_event());
-	add_filter_check(new sinsp_filter_check_event());
-	add_filter_check(new sinsp_filter_check_thread());
-	add_filter_check(new sinsp_filter_check_user());
-	add_filter_check(new sinsp_filter_check_group());
-	add_filter_check(new sinsp_filter_check_container());
-	add_filter_check(new sinsp_filter_check_fd());
-	add_filter_check(new sinsp_filter_check_fspath());
-	add_filter_check(new sinsp_filter_check_syslog());
-	add_filter_check(new sinsp_filter_check_utils());
-	add_filter_check(new sinsp_filter_check_fdlist());
-	add_filter_check(new sinsp_filter_check_k8s());
-	add_filter_check(new sinsp_filter_check_mesos());
-	add_filter_check(new sinsp_filter_check_tracer());
-	add_filter_check(new sinsp_filter_check_evtin());
+	add_filter_check(std::make_unique<sinsp_filter_check_gen_event>());
+	add_filter_check(std::make_unique<sinsp_filter_check_event>());
+	add_filter_check(std::make_unique<sinsp_filter_check_thread>());
+	add_filter_check(std::make_unique<sinsp_filter_check_user>());
+	add_filter_check(std::make_unique<sinsp_filter_check_group>());
+	add_filter_check(std::make_unique<sinsp_filter_check_container>());
+	add_filter_check(std::make_unique<sinsp_filter_check_fd>());
+	add_filter_check(std::make_unique<sinsp_filter_check_fspath>());
+	add_filter_check(std::make_unique<sinsp_filter_check_syslog>());
+	add_filter_check(std::make_unique<sinsp_filter_check_utils>());
+	add_filter_check(std::make_unique<sinsp_filter_check_fdlist>());
+	add_filter_check(std::make_unique<sinsp_filter_check_k8s>());
+	add_filter_check(std::make_unique<sinsp_filter_check_mesos>());
+	add_filter_check(std::make_unique<sinsp_filter_check_tracer>());
+	add_filter_check(std::make_unique<sinsp_filter_check_evtin>());
```

Comment:
Similar comment to the previous one on ownership, `add_filter_check` could keep accepting a raw pointer and internally turn it into a `std::unique_ptr`, it would reduce the amount of changes here. It's just a matter of understanding that whatever you provide to `add_filter_check` is now owned by it and should not be used by other pieces of the code.

Reply:
This is pretty much covered by my answer to the previous comment I think.